MultiSource/Benchmarks/Olden/em3d/make_graph.c

```diff
@@ -53,25 +55,27 @@ void fill_table(node_t **node_table, double *values, int size, int procname)
   cur_node->next = NULL;
 }
 
-void make_neighbors(node_t *nodelist, node_t **table[], int tablesz,
+void make_neighbors(ptr<node_t> nodelist, array_ptr<ptr<node_t>*>table : count(PROCS), int tablesz,
 		    int degree, int percent_local, int id)
 {
-  node_t *cur_node;
+  ptr<node_t> cur_node = nodelist;
 
-  for(cur_node = nodelist; cur_node; cur_node=cur_node->next) {
-    node_t *other_node;
+  for(; cur_node; cur_node=cur_node->next) {
+    // CHECKED C: Changed by checked-c-convert
+    // CHECKED C: manually added null initializer
+    ptr<node_t> other_node = NULL;
     int j,k;
     int dest_proc;
 
-    cur_node->to_nodes = (node_t **)malloc(degree*(sizeof(node_t *)));
+    cur_node->to_nodes = (ptr<node_t> *)malloc(degree*(sizeof(ptr<node_t>)));
     if (!cur_node->to_nodes) {
       chatting("Uncaught malloc error\n");
       exit(0);
     }
 
     for (j=0; j<degree; j++) {
       do {
-        node_t **local_table;
+        ptr<node_t> *local_table;
```

Comment:
This, again, is not the best possible type, but we later run into problems if i give it the right type of `array_ptr<ptr<node_t>> local_table : count(tablesz)`